src/renderer/components/dock/monaco-model.store.ts

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (c) 2021 OpenLens Authors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+import autoBind from "auto-bind";
+import { action, makeObservable } from "mobx";
+import {monaco} from "react-monaco-editor";
+import { createStorage } from "../../utils";
+
+export type TabId = string;
+
+interface ModelEntry {
+    id?: TabId;
+    modelUri?: monaco.Uri;
+    lang?: string;
+  }
+  
+export interface ModelsStorageState {
+    models: ModelEntry[];
+  } 
+  
+
+export class MonacoModelsStore implements ModelsStorageState {
+    private storage = createStorage<ModelsStorageState>("monaco-models", { 
+      models: new Array<ModelEntry>(),
+    });
+
+    constructor() {
+      makeObservable(this);
+      autoBind(this);
+    }
+    
+    get models(){
+      return this.storage.get().models;
+    }
+
+    set models(models: ModelEntry[]) {
+      this.storage.merge({ models });
+    }
+
+    @action
+    addModel(tabId: string, value = "", lang = "yaml"): monaco.editor.ITextModel {
```

Comment:
For three parameters, especially since they have default values. I think that this should be the following. So that new arguments can be added in the future.

```ts
function addModel(tabId: string, { value = "", lang = "yaml" } = {}): monaco.editor.ITextModel;
```

Reply:
It makes sense, thanks